c7n/query.py

```diff
@@ -540,6 +529,58 @@ def generate_arn(self):
         return self._generate_arn
 
 
+class MaxResourceLimit(object):
+
+    C7N_MAXRES_OP = os.environ.get("C7N_MAXRES_OP", 'or')
+    C7N_MAXRES_PERCENT = os.environ.get("C7N_MAXRES_PERCENT", 100)
+
+    def __init__(self, policy, selection_count, population_count):
+        self.p = policy
+        self.op = MaxResourceLimit.C7N_MAXRES_OP
+        self.selection_count = selection_count
+        self.population_count = population_count
+        self.amount = selection_count
```

Comment:
amount / percentage_amount / percent would be better initialized as none and utilized only if they are set explicitly in parse.

Reply:
will change the None. the reason i didn't do that is that it causes more if statements to check each of them since you can't do comparison operations with none objects

Reply:
i though it would just be another clause to the existing set of if statements